add Fireblast

mohtashin opened the pull request:
A self-contained Streamlit application that demonstrates Firebolt database performance through interactive load testing and DML operations monitoring.

Commit:
add Fireblast

fb-cole commented on Fireblast/README.md:
```diff
@@ -0,0 +1,345 @@
+
+## What the Demo Does
+
+### **Which Tab Should I Start With?**
```
This is a little passive. Literally nothing works if you don't start by populating your database. It's not that you _should_ start with it or that this is the _recommended_ workflow - you have to. You can be authoritative.

fb-cole commented on Fireblast/README.md:
```diff
@@ -0,0 +1,345 @@
+- ❌ **Unchecked**: Uses random generated queries (aggregations, point lookups)
+- **Best practice**: Use defined queries for realistic workload simulation
+
+**Use Cache?**:
+- ✅ **Checked**: Enables both result_cache and subresult_cache (faster responses)
+- ❌ **Unchecked**: Disables subresult_cache but keeps result_cache (tests raw performance)
```
Unchecked _should_ be the inverse behavior to what's described. There's no reason to disable the subresult cache in Firebolt, but disabling the result cache is meaningful for testing.

fb-cole commented on Fireblast/streamlit_app.py:
```diff
@@ -0,0 +1,732 @@
+            cur.execute("SET optimizer_mode=user_guided")
+
+        if not use_cache:
+            cur.execute("SET enable_result_cache = TRUE")
+            cur.execute("SET enable_subresult_cache = FALSE")
```
Yeah, so let's swap this.
```suggestion
            cur.execute("SET enable_result_cache = FALSE")
            cur.execute("SET enable_subresult_cache = TRUE")
```

fb-cole commented on Fireblast/README.md:
```diff
@@ -0,0 +1,345 @@
+- **Throughput Charts**: Operations per second over time
+- **Conflict Resolution**: Graceful handling of concurrent modification conflicts
+
+#### **💡 Best Practices:**
```
This is repetitive and feels a little like AI slop. I'm not sure any of these are great best practices.

fb-cole commented on Fireblast/README.md:
```diff
@@ -0,0 +1,345 @@
+
+---
+
+## 🔧 Troubleshooting
```
We can probably throw out the entire troubleshooting section. None of the problems described below should happen if everything is well-documented up front.

Commit:
Update Fireblast demo based on PR feedback

- Make workflow section authoritative instead of recommended
- Fix cache behavior: unchecked now disables result_cache but keeps subresult_cache
- Update README cache description to match corrected behavior

Commit:
Remove redundant cache behavior description

The unchecked behavior is now implied and doesn't need explicit documentation
